auth/saml/index.php

```diff
@@ -189,6 +189,8 @@
             $err['login'] = get_string("auth_saml_error_complete_user_data", "auth_saml", $username);
             saml_error($err['login'], '?logout', $pluginconfig->samllogfile);
         }
+
+        add_to_log(SITEID, "user", "login", "view.php?id=".$user->id."&course=".SITEID, $user->id, 0, $user->id);
```

Comment:
Hmm, I'm checking what auth_shibboleth does, and it had this call removed in moodle/moodle@5991cabfd7f34de02a3821fb71df8615846de39f, claiming that `complete_user_login()` already does this for us.

So that's weird, because I clearly remember having an empty log for several days, until adding the explicit call.

Perhaps it changed in 2.9?